src/scripts/app.js

```diff
@@ -1,15 +1,171 @@
+      }
+      if (dotsList.length % 4 != 0) {
+        dotsList.item(i - 4).classList.remove('visible');
+        dotsList.item(dotsList.length - 4).classList.add('visible');
+      }
+    } 
+  }
+
+  function myFunctionMedium(mq) {
+    if (mq.matches) { // If the viewport is more than 768px and less than 991.98px
+      var dotsList = document.querySelectorAll('.products-slider-nav ul li');
+      dotsList.forEach(function(item){ item.classList.remove('visible');});
+      //hide dots not needed for current media query
+      for (var i = 0; i < dotsList.length; i += 2) {
+        dotsList.item(i).classList.add('visible');
+      }
+      if (dotsList.length % 2 != 0) {
+        dotsList.item(i - 2).classList.remove('visible');
+        dotsList.item(dotsList.length - 2).classList.add('visible');
+      }
+    }
+  }
+
+  function myFunctionSmall(mq) {
+    if (mq.matches) { // If the viewport is 767.98px wide or less
+      document.querySelectorAll('.products-slider-nav ul li').forEach(function(item){ item.classList.add('visible');});
+    }
+  }
+
+  var smallDevice = window.matchMedia("(max-width: 767.98px)");
+  smallDevice.addListener(myFunctionSmall); // Attach listener function on state changes
+
+  var mediumDevice = window.matchMedia("(max-width: 991.98px) and (min-width: 768px)");
+  mediumDevice.addListener(myFunctionMedium); // Attach listener function on state changes
+
+  var largeDevice = window.matchMedia("(min-width: 992px)");
+  largeDevice.addListener(myFunctionLarge); // Attach listener function on state changes
+
+  //remove Bootstrap's grid for slider to work correctly
+  var removeGrid = function(){
+    var elementsWithGrid = document.getElementsByClassName('tns-item');
+    for (var i = 0; i < elementsWithGrid.length; i++) {
+      var classesToRemove = elementsWithGrid[i].classList.value.match(/\bcol-[^\s]*/g);
+      elementsWithGrid[i].classList.remove(...classesToRemove);
+    }
+  };
+
+  var slider;
+  var startSlider = function(sliderId){
+    slider = tns({
+      "container": "#" + sliderId,
+      "navContainer": ".products-slider-nav ul",
+      "controls": false,
+      "navAsThumbnails": true,
+      "items": 1,
+      "slideBy": "page",
+      "rewind": true,
+      "gutter": 27,
+      "edgePadding": 0,
+      "mouseDrag": true,
+      autoplay: true,
+      "autoplayButtonOutput": false,
+      responsive: {
+        576: {
+          items: 1,
+          "slideBy": 1
+        },
+        768: {
+          items: 2,
+          "slideBy": 2
+        },
+        992: {
+          items: 4,
+          "slideBy": 4
+        }
+      }
+    });
+    removeGrid();
+  };
+  
+  //TABS
+
+  function openTab(evt, tab) {
+    // Declare all variables
+    var i, tabcontent, tablinks;
+    document.querySelector('.panel-bar .menu ul li a').classList.remove('active');
+
+    // Get all elements with class="tabcontent" and hide them
+    tabcontent = document.getElementsByClassName("tab-pane");
```

Comment:
Gdyby gdzieś na stronie w innej sekcji znalazły się jeszcze bootstrapowe taby, to czy nie wpłynęłabyś tym kodem też na nie? Może lepiej byłoby podać nazwę sekcji, np.: `querySelectorAll(".section--products .tab-pane")`. 
To samo dotyczy linii 138. 
(Jeśli się mylę, to niech mnie ktoś poprawi :)